File: src/main/java/com/clinica/atendimento/service/consumer/ExtrairConsumer.java
package com.clinica.atendimento.service.consumer;

import com.clinica.atendimento.service.deepseek.DeepSeekService;
import com.clinica.atendimento.service.websocket.WebSocketHandler;
import org.apache.kafka.clients.consumer.ConsumerRecord;
import org.apache.kafka.common.serialization.StringDeserializer;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

@Component
public class ExtrairConsumer extends AbstractConsumer<String, String> {

    private final DeepSeekService deepSeekService;
    private final WebSocketHandler webSocketHandler;

    public ExtrairConsumer(
            @Value("${kafka.url}") String bootstrapServer,
            @Value("${kafka.topico.extrair}") String topico,
            DeepSeekService deepSeekService,
            WebSocketHandler webSocketHandler) {
        super(bootstrapServer, topico, StringDeserializer.class, StringDeserializer.class);
        this.deepSeekService = deepSeekService;
        this.webSocketHandler = webSocketHandler;
    }

    @Override
    protected void receber(ConsumerRecord<String, String> record) {
        String sessao = record.key();
        String texto = record.value();
        String extrato = deepSeekService.extrair(texto);

        webSocketHandler.sendToClientId(sessao, "/topic/transcricao/response/extrato", extrato);
    }
}

File: src/main/kotlin/com/clinica/atendimento/service/consumer/ExtrairConsumer.kt
package com.clinica.atendimento.service.consumer

import com.clinica.atendimento.service.deepseek.DeepSeekService
import com.clinica.atendimento.service.websocket.WebSocketHandler
import org.apache.kafka.clients.consumer.ConsumerRecord
import org.apache.kafka.common.serialization.StringDeserializer
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

@Component
class ExtrairConsumer(
    @Value("\${kafka.url}") bootstrapServer: String,
    @Value("\${kafka.topico.extrair}") topico: String,
    private val deepSeekService: DeepSeekService,
    private val webSocketHandler: WebSocketHandler
) : AbstractConsumer<String, String>(
    bootstrapServer,
    topico,
    StringDeserializer::class.java,
    StringDeserializer::class.java
) {

    override fun receber(record: ConsumerRecord<String, String>) {
        val sessao = record.key()
        val texto = record.value()
        val extrato = deepSeekService.extrair(texto)

        webSocketHandler.sendToClientId(sessao, "/topic/transcricao/response/extrato", extrato)
    }
}